torch/testing/_internal/common_cuda.py

```diff
@@ -32,3 +34,52 @@ def initialize_cuda_context_rng():
         for i in range(torch.cuda.device_count()):
             torch.randn(1, device="cuda:{}".format(i))
         __cuda_ctx_rng_initialized = True
+
+
+def tf32_is_not_fp32():
+    if not torch.cuda.is_available():
+        return False
+    if torch.cuda.get_device_properties(torch.cuda.current_device()).major < 8:
+        return False
+    if int(torch.version.cuda.split('.')[0]) < 11:
+        return False
+    return True
+
+
+def tf32_on_and_off(precision=1e-5):
+    def call_with_tf32_on_and_off(self, function_call):
+        old_allow_tf32 = torch.backends.cuda.matmul.allow_tf32
+        old_precison = self.precision
+        try:
+            torch.backends.cuda.matmul.allow_tf32 = False
+            function_call()
+            torch.backends.cuda.matmul.allow_tf32 = True
+            self.precision = precision
+            function_call()
+        finally:
+            torch.backends.cuda.matmul.allow_tf32 = old_allow_tf32
+            self.precision = old_precison
+
+    def wrapper(f):
+        nargs = len(inspect.signature(f).parameters)
+        if nargs == 2:
+            @functools.wraps(f)
+            def wrapped(self, device):
+                assert isinstance(device, str)
```

Comment:
You don't need this assert once you switch to using self.device_type

Reply:
fixed